src/utils/wallet.js

```diff
@@ -148,13 +160,20 @@ export function hashTypedDataMessage(message: string): string {
 }
 
 // handle eth_signTypedData
-export function signTypedData(data: string, walletInstance: Object): Promise<string> {
+export function signTypedData(
+  data: string,
+  walletInstance: Object,
+  isLegacyEIP1271: boolean = false, // EIP-1271 had few proposals that went live
+): Promise<string> {
   const provider = getEthereumProvider(getEnv().NETWORK_PROVIDER);
   const wallet = walletInstance.connect(provider);
 
-  const hashedTypedData = hashTypedDataMessage(data);
+  if (isLegacyEIP1271) {
+    const hashedTypedData = hashTypedDataMessage(data);
+    return wallet.signMessage(ethers.utils.arrayify(hashedTypedData));
+  }
 
-  return wallet.signMessage(ethers.utils.arrayify(hashedTypedData));
+  return new Promise((resolve) => resolve(signTypedData_v4(toBuffer(wallet.privateKey), { data: JSON.parse(data) })));
```

Comment:
Removed this 👍 

Reason here if it fails is to catch it with `.catch` instead of `try...catch` block and I had this unified promise handler implementation which I replaced later, but left promise wrapper here.